infrastructure/network/addressmanager/addressmanager.go

```diff
@@ -111,7 +117,7 @@ func (am *AddressManager) RemoveAddress(address *appmessage.NetAddress) {
 
 	key := netAddressKey(address)
 	delete(am.addresses, key)
-	delete(am.bannedAddresses, key)
+	delete(am.bannedAddresses, key.address)
```

Comment:
Banned addresses should remain banned until they are unbanned. Please remove this line.